### DISCUSSION TITLE
Will Turbo Cache be used when changes occur?

### DISCUSSION THREAD

**Discussion by woobin-finda** [vercel/turborepo#9171] [Help] [answered]
### Summary

this is my  package.json:
```
{
  "scripts": {
     "build": "turbo run build --filter=repo --cache-dir='./apps/repo/.turbo/cache'",
  }
}
```
After building, if you build again as is, the cache will be applied:
```
Tasks:    1 successful, 1 total
Cached:    1 cached, 1 total
Time:    376ms >>> FULL TURBO
```

But when i change small occurs, they say "cache miss, ...":
```
• Packages in scope: repo
• Running build:stg in 1 packages
• Remote caching disabled
repo:build: cache miss, executing...
```
At first, I thought it was a next cache issue and cached the next cache as well, but as small changes continue to occur, that sentence appears.
When I run it, it builds about 10 to 20 seconds faster than the first build. If so, is the turbo cache being used correctly? I'm also curious what the principle is.

thanks.

**Comment by anthonyshew** [comment, marked as answer]
👋 A change to a package will result in a cache miss for that package. Caching is done per-package, so changing any file in the Next.js application will cause it to miss Turborepo cache.

Separately, Next.js has its own caching in the `.next` directory, which likely accounts for the faster subsequent builds to that app.
